sources/packages/backend/libs/integrations/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt.processing.service.ts

```diff
@@ -50,7 +50,7 @@ export class DisbursementReceiptProcessingService {
   async process(auditUserId: number): Promise<ProcessSFTPResponseResult[]> {
     // Get the list of all files from SFTP ordered by file name.
     const fileSearch = new RegExp(
-      `^${this.esdcConfig.environmentCode}EDU.PBC.DIS.D[\w]*\.[0-9]*`,
+      `^${this.esdcConfig.environmentCode}EDU.PBC.DIS.[\\w]*\\.[0-9]*`,
```

Comment:
The intermediate points should also be escaped:
```
^${this.esdcConfig.environmentCode}EDU\\.PBC\\.DIS\\.[\\w]*\\.[0-9]*
```
Now that the format is well known, a more precise regex could be used, as below.
```
^${this.esdcConfig.environmentCode}EDU\\.PBC\\.DIS.[0-9]{8}\\.[0-9]{3}$
```

Reply:
Thanks @andrewsignori-aot for the precise regex format. The PR is updated with this regex.

Reply:
It looks good, minor observation that the `$` at the end was part of it, but it would not be a blocker 😉

Reply:
The anchor tag is now added. This `$` works 🙂